back-end/application.js

```diff
@@ -0,0 +1,59 @@
+// load .env data into process.env
+// require("dotenv").config();
+
+// Web server config
+const express = require("express");
+const bodyParser = require("body-parser");
+const sass = require("node-sass-middleware");
+const cors = require("cors");
+const morgan = require("morgan");
+const session = require("express-session");
+const app = express();
+
+const registerRoutes = require("./src/routes/register");
+const loginRoutes = require("./src/routes/login");
+const editProfileRoutes = require("./src/routes/editProfile");
+const questRoutes = require("./src/routes/quests");
+const userRoutes = require("./src/routes/users");
+const badgeRoutes = require("./src/routes/badges");
+const classRoutes = require("./src/routes/classes");
+
+module.exports = function application(ENV) {
+  if (ENV !== "test") {
+    require("dotenv").config();
+  }
```

Comment:
I think it would be better if nowhere in the code we would call `require("dotenv").config()` directly, _except_ from the `environment.js` file.

That way there would only be a single place to deal with when it comes to load the ENV vars. That would make it easier to maintain/debug/read.

Now you might wonder how would the ENV vars be loaded for the development environment? Having that line on top of `back-end/server.js` would take care of it:

```js
require("./environment"); 
```